src/util.py

```diff
@@ -21,8 +21,9 @@ def datetime_from_isoformat(s):
 
     # fallback: use strptime. Problem is timezone with colon
     # Thanks SO! (https://stackoverflow.com/questions/30999230/how-to-parse-timezone-with-colon)
-    if s[-3:-2] == ':':
-        s = s[:-3]+s[-2:]
+    if s[-6:-5] == "+":
```

Comment:
datetimes in SpiceEV should always be timezone-aware. Not sure if we should support timezone-agnostic parsing in an outdated Python version. Anyway, this can be simplified:
```
if s[-6] == "+" and s[-3] == ":":
    s = s[:-3] + s[-2:]
```